pkg/infra/util/util.go

```diff
@@ -74,3 +77,23 @@ func GenerateUID(nn types.NamespacedName) string {
 	hashutil.DeepHashObject(hash, nn.String())
 	return rand.SafeEncodeString(fmt.Sprint(hash.Sum32()))
 }
+
+func AdditionalResources(infraCluster *infrav1.Cluster) []addonsv1.ResourceRef {
+	refs := make([]addonsv1.ResourceRef, 0, len(infraCluster.Spec.AdditionalResources))
+	for _, resource := range infraCluster.Spec.AdditionalResources {
+		refs = append(refs, addonsv1.ResourceRef{
+			Kind: resource.Kind,
+			Name: resource.Name,
+		})
+	}
+
+	sort.Slice(refs, func(i, j int) bool {
+		if refs[i].Kind == refs[j].Kind {
+			return refs[i].Name < refs[j].Name
+		}
+
+		return refs[i].Kind < refs[j].Kind
```

Comment:
I donot think we need to sort again, this is already an array specified by enduser, if the resources has  dependency it would fail if we change the order